dozer-api/src/grpc/internal/internal_pipeline_server.rs

```diff
@@ -11,13 +11,13 @@ use dozer_types::grpc_types::internal::{
 use dozer_types::log::info;
 use dozer_types::models::api_config::AppGrpcOptions;
 use dozer_types::models::api_endpoint::ApiEndpoint;
+use dozer_types::parking_lot::Mutex;
```

Comment:
The root cause is very remote. If you check [here](https://github.com/chubei/dozer/blob/feat/checkpoint/dozer-log/src/storage/queue.rs#L64), I'm using `blocking_send` to make `send_request` a `sync fn` instead of `async fn`. I'm doing this because this method is going to be called by `Processor::persist` which I plan to keep as a `sync fn`.

The causal chain of this `Mutex` change is like this:

- `blocking_send` must be called outside of tokio context.
- `Log::persist` must be called outside of tokio context. 
- `Log` has to be put inside a `parking_lot::Mutex` to be locked in `LogSink::persist` outside of tokio context.
- Internal pipeline server shares the `Arc<Mutex<Log>>` so it gets changed to `parking_lot::Mutex`.

Do you think making `send_request` a `sync fn` is a bad idea? Should we make `Processor` and `Sink` `async_trait` and their persist methods `async`?

Reply:
Right I think so too. So which solution do you prefer? Or any other ideas?

Reply:
Actually I prefer to fix it now. It's not a big change once we decide what to do. I can create another PR in no time.

Reply:
Sure. I'll make `Processor::persist` and `Sink::persist` async then.